### PR TITLE
Changing overlay/extension merging algorithm

### PR DESCRIPTION
**Pull request by petrochenko-pavel-a**
Suggested changes based on investigation of https://github.com/raml-org/raml-js-parser-2/issues/177 and according to #488

### CODE REVIEW

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -3516,6 +3518,8 @@ For each **Current Extension Tree Object property** the following is done:
   * All _Conflicting Properties_ are removed from the **Current Target Tree Object**
   * The **property** is added to the **Current Target Tree Object**.
 
+Traits and Resource Types applications are appled one more time to the _Target Tree_.
```
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: appled -> applied

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -3492,6 +3492,8 @@ All _includes_ are resolved and applied for both _Master Tree_ and _Extension Tr
 
 All _uses_ are resolved and applied for both _Master Tree_ and _Extension Tree_. The trees MUST NOT have _uses_ instructions with the same namespace referring to different files.
 
+All Trait and Resource Types applications are aplied in the _Master Tree_
```
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: aplied -> applied

**Comment by sichvoge** [comment]
Seems to be only a small change to the spec. After @petrochenko-pavel-a fixed the spelling mistakes: LGTM

**Comment by usarid** [comment]
LGTM

**Pull request merged**